azure-mgmt-scheduler/src/main/java/com/microsoft/azure/management/scheduler/Job.java

```diff
@@ -93,33 +98,217 @@
          */
         interface WithStartTime {
             /**
+             * A job which will start right away.
+             *
+             * @return the next stage of the definition
+             */
+            WithRecurrenceOrAction startingNow();
+
+            /**
              * Specifies the start time of the job.
              *
              * @param startTime the start time
              * @return the next stage of the definition
              */
-            WithRecurrenceOrAction withStartTime(DateTime startTime);
+            WithRecurrenceOrAction startingAt(DateTime startTime);
         }
 
         /**
          * The stage of the job definition for an Azure Scheduler allowing to specify the recurrence and action details.
          */
-        interface WithRecurrenceOrAction {
+        interface WithRecurrenceOrAction extends WithAction {
             /**
-             * Specifies the recurrence  of the job.
+             * The recurrence object specifies recurrence rules for the job and the recurrence the job will execute with.
+             * <p>
+             * The recurrence object supports the elements frequency, interval, endTime, count, and schedule. If recurrence
+             *   is defined, frequency is required; the other elements of recurrence are optional.
              *
-             * @param jobRecurrence the start time
+             * @param jobRecurrence the reccurence object
              * @return the next stage of the definition
              */
             WithAction withRecurrence(JobRecurrence jobRecurrence);
 
             /**
-             * Specifies the action details of the job.
+             * Specifies the interval for the frequency that determines how often the job will run.
```

Comment:
This description seems to use the Swagger's confusing naming, which even I have a hard time following. It'd be good to explain this in simpler terms. How about something like: 
```
Specifies the number of time units between consecutive occurrences. The exact time unit to use is specified at the next stage.
<p>
For example, if 3 is specified here and "week" is selected at the next stage, the job will recur every 3 weeks.
<p>
Different time units support different maximum numbers of occurrences. Those maximum combinations are: 18 months, 78 weeks, 548 days, 1000 hours or minutes.
```